core/src/com/unciv/ui/screens/worldscreen/WorldMapHolder.kt

```diff
@@ -712,13 +712,11 @@ class WorldMapHolder(
 
         // Highlight best tiles for city founding
         if (unit.hasUnique(UniqueType.FoundCity)
-                && UncivGame.Current.settings.showSettlersSuggestedCityLocations
-        ) {
-            CityLocationTileRanker.getBestTilesToFoundCity(unit).map { it.first }
-                .filter { it.isExplored(unit.civ) }.take(3).forEach {
-                    tileGroups[it]!!.layerOverlay.showGoodCityLocationIndicator()
-                }
-
+                && UncivGame.Current.settings.showSettlersSuggestedCityLocations) {
+            CityLocationTileRanker.getBestTilesToFoundCity(unit).first.asSequence()
+                .sortedByDescending { it.value }.filter { it.key.isExplored(unit.civ) }.take(3).forEach {
```

Comment:
Filter first, sort later? I don't think it really matters, but if we're already changing this...